### PR TITLE
Shaokai/sa finetune

### PR DESCRIPTION
**Pull request by yeshaokai**
Added SA fine-tuning code
Added a new example project for testing fine-tuning
Added a testscript

### CODE REVIEW

File: deeplabcut/generate_training_dataset/trainingsetmanipulation.py [hunk 1]
```diff
@@ -728,6 +729,7 @@ def create_training_dataset(
     net_type=None,
     augmenter_type=None,
     posecfg_template=None,
+    superanimal_name = ""
```
Comment (jeylau) [inline deeplabcut/generate_training_dataset/trainingsetmanipulation.py]: Is this argument meant to be user-facing? Or only for internal purposes? It is very redundant with posecfg_template.

File: deeplabcut/pose_estimation_tensorflow/training.py [hunk 2]
```diff
@@ -190,7 +194,54 @@ def train_network(
...
+                / (superanimal_name + "_weights")
+            )    
+
+            if superanimal_name in MODELOPTIONS:
+                if not os.path.exists(weight_folder):
+                    download_huggingface_model(superanimal_name, weight_folder)
+                else:
+                    print(f"{weight_folder} exists, using the downloaded weights")
+            else:
+                print(f"{superanimal_name} not available. Available ones are: ", MODELOPTIONS)
```
Comment (jeylau) [inline deeplabcut/pose_estimation_tensorflow/training.py:222]: We could move the imports higher up, and test early whether `superanimal_name` is in `MODELOPTIONS`.

File: deeplabcut/pose_estimation_tensorflow/training.py [hunk 3]
```diff
@@ -127,6 +128,9 @@ def train_network(
         Directory containing the deeplabcut models to use when evaluating the network.
         By default, the models are assumed to exist in the project folder.
 
+    superanimal_name: str, optional, default =""
+        Specified if suepranimal fine-tuning is needed
```
Comment (jeylau) [inline deeplabcut/pose_estimation_tensorflow/training.py]:
```suggestion
        Specified if superanimal fine-tuning is needed
```

**Review by jeylau — COMMENTED** [review]
Looks fine to me! I guess we'll get rid of the extra data in examples/

File: deeplabcut/generate_training_dataset/trainingsetmanipulation.py (hunk 1 above)
Comment (yeshaokai) [reply deeplabcut/generate_training_dataset/trainingsetmanipulation.py]: I am not sure it was me who put it there. Is it used anywhere in the source code?

Comment (jeylau) [reply deeplabcut/generate_training_dataset/trainingsetmanipulation.py]: It's not; I'll get rid of it.

**Comment by MMathisLab** [comment]
We need the superanimal name... it was discussed and reviewed and in other code with @n-poulsen

**Comment by MMathisLab** [comment]
closing as redundant with #2389

**Pull request closed without merging**